xbmc/windowing/X11/WinSystemX11GLContext.cpp

```diff
@@ -175,18 +180,25 @@ bool CWinSystemX11GLContext::DestroyWindow()
 
 XVisualInfo* CWinSystemX11GLContext::GetVisual()
 {
-  GLint att[] =
+  if(!m_pGLContext)
   {
-    GLX_RGBA,
-    GLX_RED_SIZE, 8,
-    GLX_GREEN_SIZE, 8,
-    GLX_BLUE_SIZE, 8,
-    GLX_ALPHA_SIZE, 8,
-    GLX_DEPTH_SIZE, 24,
-    GLX_DOUBLEBUFFER,
-    None
-  };
-  return glXChooseVisual(m_dpy, m_nScreen, att);
+    m_pGLContext = new CGLContextEGL(m_dpy);
+  }
+
+#ifdef HAS_GLX
```

Comment:
@lrusak this entire block should be enclosed by if(!m_pGLContext)
currently it kills the context every time we ask for a visual